### ISSUE TITLE
请问为什么l_target 和 l_detail 的损失求解要放在torch.no_grad()部分呢？

### ISSUE THREAD

**Issue by linklist2** [JinyuanLiu-CV/TarDAL#4]
with torch.no_grad():
            self.dis_target.eval()
            l_target = -self.dis_target(fus * mk).mean()  # judge target: fus * m
            self.dis_detail.eval()
            l_detail = -self.dis_detail(self.gradient(fus * (1 - mk))).mean()  # judge detail: Grad(fus * (1-mk))

        c1, c2 = self.config.adv_weight  # c1 * l_target + c2 * l_detail
        l_adv = c1 * l_target + c2 * l_detail

        loss = l_src + b3 * l_adv

您好，首先非常感谢您的工作，但是能否请问在[pipeline/train.py](url)出现的train_generator，为什么l_target 和 l_detail 的损失求解要放在torch.no_grad()部分呢？
torch.no_grad()中貌似不会得到任何反向传播的梯度的。

**Comment by MisakiCoca** [comment]
感谢您对我们工作的关注！
在论文实验使用的代码版本中，该处实现使用了梯度。在开源整理时发生了一些笔误，现已改正，非常感谢您的指正！
目前仓库内代码的训练部分仅为过渡版本，我们将在一个月内发布具有目标检测联合损失的完整版本。

![file_info](https://user-images.githubusercontent.com/40165523/169683530-0a83f0d6-3284-430f-ad98-bd9c78fa23f8.png)
![code_shot](https://user-images.githubusercontent.com/40165523/169683845-1802ef98-b601-4df9-aa49-9cc514cab2ee.png)
